components/prism-core.js

```diff
@@ -248,62 +312,64 @@ var _ = self.Prism = {
 			}
 
 			var patterns = grammar[token];
-			patterns = (_.util.type(patterns) === "Array") ? patterns : [patterns];
 
-			for (var j = 0; j < patterns.length; ++j) {
-				var pattern = patterns[j],
-					inside = pattern.inside,
-					lookbehind = !!pattern.lookbehind,
-					lookbehindLength = 0,
-					alias = pattern.alias;
+			patterns = (_.util.type(patterns) === "Array") ? patterns : patterns.competition ? grammar._competition : [patterns];
```

Comment:
Yes that is also an error. Sorry for that. I first implemented the competition inside of arrays and then patched on the global competition without enough testing as it seems. I have to think about how to solve this. Thanks for your review!